tests/conftest.py

```diff
@@ -56,23 +64,58 @@ def auth_state(MockAuthClient, config, monkeypatch) -> AuthState:
     return auth_state
 
 
+@pytest.fixture(scope="session", autouse=True)
+def register_api_fixtures():
+    for yaml_file in (pathlib.Path(__file__).parent / "api-fixtures").rglob("*.yaml"):
+        response_set = yaml.safe_load(yaml_file.read_text())
+        register_response_set(yaml_file.stem, response_set)
+
+
+@pytest.fixture(autouse=True)
+def responses() -> responses_module.RequestsMock:
+    """Mock all requests.
+
+    The default `responses.mock` object is returned,
+    which allows tests to access various properties of the mock.
+    For example, they might check the number of intercepted `.calls`.
+    """
+
+    responses_module.reset()
+    responses_module.start()
+    try:
+        yield responses_module.mock
+    finally:
+        responses_module.stop()
+
+
 @pytest.fixture
-def duplicate_auth_state(auth_state: AuthState, config) -> AuthState:
-    """This fixture provides an AuthState type object that has the same token value as the
-    auth_state fixture above. This means that caching should make access to either of the
-    returned auth_state objects return values from the same cache as the root key is the
-    token value.
+def freeze_time() -> t.Generator[
+    t.Callable[[RegisteredResponse], RegisteredResponse], None, None
+]:
+    """Inspect a Globus SDK RegisteredResponse object and freeze time if needed.
 
+    Some responses may only be valid within a specific time range
+    (for example, a token may only be valid within a specific time period).
+    This fixture creates a function that will look for a "freezegun" key
+    in the Response metadata. If found, time will be frozen at that value.
     """
 
-    # Create a TokenChecker to be used to create a mocked auth_state object
-    checker = TokenChecker(
-        client_id=config["client_id"],
-        client_secret=config["client_secret"],
-        expected_scopes=config["expected_scopes"],
-        expected_audience=config["expected_audience"],
-    )
-    dup_auth_state = checker.check_token(auth_state.bearer_token)
-    # Set up the same mocked Auth client
-    dup_auth_state.auth_client = auth_state.auth_client
-    return dup_auth_state
+    frozen_time: t.Optional[freezegun.freeze_time] = None
+
+    def freezer(response: RegisteredResponse) -> RegisteredResponse:
+        """Freeze time based on a "freezegun" key (if any) in the response metadata."""
+
+        if "freezegun" in response.metadata:
+            instant = datetime.datetime.utcfromtimestamp(response.metadata["freezegun"])
+            # Update `frozen_time` in the outer scope.
+            nonlocal frozen_time
+            frozen_time = freezegun.freeze_time(instant)
```

Comment:
Perhaps worth noting: I saw this as well as a possible outcome. `frozen_time` would get set multiple times and only "unwound" once at teardown.
The testsuite itself doesn't do that so I thought it was a fine possibility to ignore for now. But maybe we should hash it out -- does `freezegun` only track the last time or does it keep a stack?

Reply:
A future test author can...er...have to finagle their timestamps to make everything happy.

#punt :foot: :football: :arrow_right:

Reply:
I'm going to mark this as resolved as this will need to be accounted for as needed.

Reply:
GitHub didn't refresh so I could see your comment as well, @sirosen. I'll adjust this to be more explicit so things are very much at the forefront. Probably an `assert` that `frozen_time` is None before doing anything else.

Reply:
I do think it'd be useful since you're introducing novel functionality here to either:
1. Make it explicit that that isn't supported (eg. throw an exception if someone tries to freeze time but it's already frozen cause the variable is set)
2. Use the [context-manager version of freeze_time](https://github.com/spulec/freezegun#context-manager) instead so that we can punt this question up to the library instead of to our future selves (`with freeze_time(response.metadata["freezegun"]): yield`)

Reply:
I've added an `assert` with the failure message "You can't freeze time twice!", and a test to verify that the `assert` can be triggered.